distributed/scheduler.py

```diff
@@ -2086,6 +2091,16 @@ def _transition(
             return recommendations, client_msgs, worker_msgs
         except Exception:
             logger.exception("Error transitioning %r from %r to %r", key, start, finish)
+            self.log_event(
+                "transitions",
+                {
+                    "action": "scheduler-transition-failed",
+                    "key": key,
+                    "start": start,
+                    "finish": finish,
+                    "transistion_log": self.transition_log.copy(),
```

Comment:
nit:

Instead of copying, I'd convert this into a type that's easier to serialize into various messaging formats, e.g., JSON.

```suggestion
                    "transistion_log": list(self.transition_log),
```